core/whistle_transactions-1.0.0/src/wh_transactions.erl

```diff
@@ -131,132 +142,173 @@ filter_by_reason(<<"only_calls">>, Transactions) ->
+-spec fetch(ne_binary(), gregorian_seconds(), gregorian_seconds()) ->
+                   {'ok', wh_transactions()} |
+                   {'error', _}.
+fetch(Account, From, To) ->
+    case check_range(Account, From, To) of
+        {'error', _R}=Error -> Error;
+        {'ok', ViewOptionsList} -> fetch(Account, ViewOptionsList)
     end.
 
-%%--------------------------------------------------------------------
 %% @private
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
--spec check_range(integer(), integer()) ->
-                         {'ok', 'undefined' | wh_proplist(), wh_proplist()} |
+-spec fetch(ne_binary(), wh_proplists()) ->
+                   {'ok', wh_transactions()} |
+                   {'error', _}.
+fetch(Account, ViewOptionsList) ->
+    case {fetch_local(Account, ViewOptionsList)
+          ,fetch_bookkeeper(Account, ViewOptionsList)
+         }
+    of
+        {{'error', _R}=Error, _} -> Error;
+        {_, {'error', _R}=Error} -> Error;
+        {{'ok', Local}, {'ok', Bookkeeper}} ->
+            {'ok', de_duplicate_transactions(Local, Bookkeeper)}
+    end.
+
+%% @private
+-spec check_range(ne_binary(), gregorian_seconds(), gregorian_seconds()) ->
+                         {'ok', wh_proplists()} |
                          {'error', ne_binary()}.
-check_range(From, To) ->
-    {{YearFrom, MonthFrom, _}, _} = calendar:gregorian_seconds_to_datetime(From),
-    {{YearTo, MonthTo, _}, _} = calendar:gregorian_seconds_to_datetime(To),
-    ViewOptionsFrom = [{'startkey', From}
-                       ,{'endkey', To}
-                       ,{'year', YearFrom}
-                       ,{'month', MonthFrom}
-                       ,'include_docs'
-                      ],
-    ViewOptionsTo = [{'startkey', From}
-                      ,{'endkey', To}
-                      ,{'year', YearTo}
-                      ,{'month', MonthTo}
-                      ,'include_docs'
-                    ],
-    case {YearTo - YearFrom, MonthTo - MonthFrom} of
-        {0, 0} -> {'ok', 'undefined', ViewOptionsTo};
-        {0, M} when M > 2 ->
-            {'error', <<"max range 2 consecutive month">>};
-        {0, _M} ->
-            {'ok', ViewOptionsFrom, ViewOptionsTo};
-        {1, -11} ->
-            {'ok', ViewOptionsFrom, ViewOptionsTo};
-        {1, _M} ->
-            {'error', <<"max range 2 consecutive month">>};
-        {_Y, _M} ->
-            {'error', <<"max range 2 consecutive month">>}
+check_range(Account, From, To) ->
+    ViewOptionsList =
+        [ begin
+              {Account, Year, Month} = kazoo_modb_util:split_account_mod(MODb),
+              [{'startkey', From}
+               ,{'endkey', To}
+               ,{'year', Year}
+               ,{'month', Month}
+               ,'include_docs'
+              ]
+          end || MODb <- kazoo_modb:get_range(Account, From, To)
+        ],
+    case length(ViewOptionsList) =< 3 of
```

Comment:
I wouldn't worry about this check; `cb_modules_util:range_view_options/2` enforces this limit. The core lib shouldn't be concerned with it.